edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Service/Program.cs

```diff
@@ -163,14 +164,29 @@ static async Task<EdgeHubProtocolHead> GetEdgeHubProtocolHeadAsync(ILogger logge
                 protocolHeads.Add(new HttpProtocolHead(hosting.WebHost));
             }
 
+            var orderedProtocolHeads = new List<IProtocolHead>();
+            if (configuration.GetValue("mqttBrokerSettings:enabled", true))
+            {
+                orderedProtocolHeads.Add(container.Resolve<MqttBrokerProtocolHead>());
+            }
+
             if (configuration.GetValue("authAgentSettings:enabled", true))
             {
-                protocolHeads.Add(await container.Resolve<Task<AuthAgentProtocolHead>>());
+                orderedProtocolHeads.Add(await container.Resolve<Task<AuthAgentProtocolHead>>());
             }
 
-            if (configuration.GetValue("mqttBrokerSettings:enabled", true))
+            switch (orderedProtocolHeads.Count)
             {
-                protocolHeads.Add(container.Resolve<MqttBrokerProtocolHead>());
+                case 0:
+                    break;
+
+                case 1:
```

Comment:
this can be just an if orderedProtocolHeads.Count >= 1 add to the list, it should not matter if the orderedprotocolheads only contain one item because it will just work to start it

Reply:
Yes, I thought the same, that the OrderedProtocolHead works 1 (or even zero) items in it - but then I thought to use it only when it actually makes something, otherwise it falls back to the original implementation.